presto-main/src/main/java/com/facebook/presto/execution/QueryExecution.java

```diff
@@ -45,7 +50,7 @@ Duration waitForStateChange(QueryState currentState, Duration maxWait)
 
     Session getSession();
 
-    void start(Optional<String> resourceGroupName);
```

Comment:
Oh the actual reason to not pass it in start is because we want to set the resource group as early as possible, rather than when it starts to run. I might take a different approach.